web_widget_numeric_step/static/src/js/numeric_step.js

```diff
@@ -133,10 +133,8 @@ odoo.define('web_widget_numeric_step.field', function (require) {
          * @param {String} mode can be "plus" or "minus"
          */
         _doStep: function (mode) {
-            var cval = Number(this.$input.val());
-            if (_.isNaN(cval)) {
-                return;
-            }
+            var field = this.record.fields[this.name];
+            var cval = field_utils.parse[field.type](this.$input.val())
```

Comment:
Hi . Thanks for this fix. using field_utils is a good idea. But don't you have to keep the like if isNan ? Because, field_utils can return a not a Number, if the input is invalid. Don't you think ? 

kind regards.

In other word, my proposal is : 
```
            var field = this.record.fields[this.name];
            var cval = field_utils.parse[field.type](this.$input.val())
            if (_.isNaN(cval)) {
                return;
            }
```

Reply:
Hi @legalsylvain the 'parse' method throws and exception if can't do the job... 
I know that this is too ugly for final user, perhaps best something like:
```
var cval = 0;
try {
    cval = field_utils.parse[field.type](this.$input.val());
} catch {
    return;
}
```

What do you think?

Reply:
For me the previous value should be put in case of error.

Reply:
Ok, updated. Now if the user input a invalid value and do an "step" call, the method will put the previous value.